test/brightIdFaucet.js

```diff
@@ -1,29 +1,499 @@
+
+          await faucetUserClaimFunction()
+
+          const ethBalanceAfter = toBn(await web3.eth.getBalance(faucetUser))
+          const tokenBalanceAfter = await token.balanceOf(faucetUser)
+          assert.isTrue(ethBalanceAfter.gt(MIN_ETH_BALANCE), 'Incorrect final eth balance, too low')
+          assert.isTrue(tokenBalanceAfter.gt(tokenBalanceBefore), 'Incorrect token balance, too low')
+          await web3.eth.sendTransaction({ from: faucetUserSecondAddress, to: faucetUser, value: ethBalanceToRemove })
+        })
+
+        it('converts faucet token to eth when used by multiple users with less than minimum eth balance', async () => {
+          const secondUserAddresses = [secondFaucetUser]
+          const secondUserSig = getVerificationsSignature(secondUserAddresses, timestamp)
+          await brightIdFaucet.claimAndOrRegister(BRIGHT_ID_CONTEXT, secondUserAddresses, timestamp, secondUserSig.v, secondUserSig.r, secondUserSig.s,
+            { from: secondFaucetUser })
+          await brightIdFaucet.mockIncreaseTime(PERIOD_LENGTH)
+          const eth1Balance = toBn(await web3.eth.getBalance(faucetUser))
+          const eth1BalanceToRemove = eth1Balance.sub(MIN_ETH_BALANCE.sub(toBnWithDecimals(2, 17))) // All - 0.3 ETH
+          await web3.eth.sendTransaction({ from: faucetUser, to: faucetUserSecondAddress, value: eth1BalanceToRemove })
+
+          const eth2Balance = toBn(await web3.eth.getBalance(secondFaucetUser))
+          const eth2BalanceToRemove = eth2Balance.sub(MIN_ETH_BALANCE.sub(toBnWithDecimals(2, 17))) // All - 0.3 ETH
+          await web3.eth.sendTransaction({
+            from: secondFaucetUser,
+            to: faucetUserSecondAddress,
+            value: eth2BalanceToRemove
+          })
+          const eth2BalanceBefore = toBn(await web3.eth.getBalance(secondFaucetUser))
+          const token2BalanceBefore = await token.balanceOf(secondFaucetUser)
+          assert.isTrue(eth2BalanceBefore.lt(MIN_ETH_BALANCE), 'Incorrect original eth balance, too high')
+
+          await faucetUserClaimFunction()
+          await secondUserClaimFunction()
+
+          const eth2BalanceAfter = toBn(await web3.eth.getBalance(secondFaucetUser))
+          const token2BalanceAfter = await token.balanceOf(secondFaucetUser)
+          assert.isTrue(eth2BalanceAfter.gt(MIN_ETH_BALANCE), 'Incorrect final eth balance, too low')
+          assert.isTrue(token2BalanceAfter.gt(token2BalanceBefore), 'Incorrect token balance, too low')
+          await web3.eth.sendTransaction({ from: faucetUserSecondAddress, to: faucetUser, value: eth1BalanceToRemove })
+          await web3.eth.sendTransaction({
+            from: faucetUserSecondAddress,
+            to: secondFaucetUser,
+            value: eth2BalanceToRemove
+          })
+        })
+
+        it('converts faucet token to eth when exchange will cost more than faucet payment', async () => {
```

Comment:
Might be a bit clearer

```suggestion
        it('converts full faucet token payment to eth when exchange will cost more than faucet payment', async () => {
```